cpu/include/Dialect/TritonCPU/IR/TritonCPUOps.td

```diff
@@ -38,6 +39,50 @@ def TTC_BlockIdOp : TTC_Op<"block_id", [Pure]> {
   }];
 }
 
+def TTC_CurrentBlockOp : TTC_Op<"current_block", [SameOperandsAndResultType]> {
+  let summary = "The current block ID for this warp.";
+
+  let description = [{
+    Get the current block ID within the loop over blocks for this persistent kernel.
+  }];
+
+  let arguments = (ins AnyType:$input);
+
+  let results = (outs AnyType:$result);
+
+  let assemblyFormat = [{
+    $input attr-dict `:` type($result)
+  }];
+}
+
+def TTC_BlockStartOp : TTC_Op<"block_start", [Pure]> {
```

Comment:
Could we just add the new parameters with a `block_start` or `block_end` attribute instead of these ops.